### PR TITLE
unify coord usage to util.Position. closes #386

### LINKED ISSUE #386: Ugliness: coordinates everywhere
rabbitescape.engine.textworld.Chars.Point
rabbitescape.engine.textworld.LineProcessor.Point
rabbitescape.engine.WorldChanges.Point
rabbitescape.engine.util.Position

### PR DESCRIPTION
**Pull request by colonelfazackerley**

### CODE REVIEW

File: rabbit-escape-engine/src/rabbitescape/engine/util/Position.java [hunk 1]
```diff
@@ -15,4 +15,36 @@ public Position plus( Position p )
...
+    @Override
+    public int compareTo( Position other )
+    {
+        if ( y < other.y )
+        {
+            return -1;
+        }
+        else if( y > other.y )
+        {
+            return 1;
```
Comment (tttppp) [inline rabbit-escape-engine/src/rabbitescape/engine/util/Position.java]: `y - other.y` would cover both not equal cases.

File: rabbit-escape-engine/src/rabbitescape/engine/util/Position.java [hunk 2]
```diff
@@ -15,4 +15,36 @@ public Position plus( Position p )
...
+            {
+                return -1;
+            }
+            else if( x > other.x )
+            {
+                return 1;
+            }
+            else
+            {
+                return 0;
```
Comment (tttppp) [inline rabbit-escape-engine/src/rabbitescape/engine/util/Position.java]: `x-other.x` would cover all three cases.

**Comment by tttppp** [comment]
Added a couple of comments on some code I only just noticed wasn't written in this change. Woot. Change looks good to me!

**Comment by colonelfazackerley** [comment]
It occurred to me too, but I left it alone.

**Comment by colonelfazackerley** [comment]
I pushed a commit that simplifies compareTo().

**Comment by colonelfazackerley** [comment]
I read the API spec for comparable, and added some more tests, and an equals for Position.

File: rabbit-escape-engine/src/rabbitescape/engine/util/Position.java [hunk 3]
```diff
@@ -15,4 +15,43 @@ public Position plus( Position p )
...
+        {
+            return false;
+        }
+        Position p = (Position)o;
+        return this.x == p.x && this.y == p.y;
+    }
+
+    /**
+     * This may break if we have levels with a dimension greater than 1000
+     * and the class is used for nominal pixels relative to the world origin.
```
Comment (tttppp) [inline rabbit-escape-engine/src/rabbitescape/engine/util/Position.java]: It won't actually break as such. It's just optimised for x coordinates between 0 and 999. Anyway, good idea to add hash code and equals!

**Comment by colonelfazackerley** [comment]
I see. Hash code collisions do not constitute an error. I will amend that comment.

**Comment by colonelfazackerley** [comment]
done

**Comment by colonelfazackerley** [comment]
i added equals, as the API spec for comparable suggests that compareto should return zero when equals returns true.

**Comment by andybalaam** [comment]
I like compareTo particularly.

**Pull request merged**